packages/hub/src/hub.ts

```diff
@@ -588,9 +695,9 @@ function getHubFromActiveDomain(registry: Carrier): Hub {
     }
 
     // If there's no hub on current domain, or it's an old API, assign a new one
-    if (!hasHubOnCarrier(activeDomain) || getHubFromCarrier(activeDomain).isOlderThan(API_VERSION)) {
-      const registryHubTopStack = getHubFromCarrier(registry).getStackTop();
-      setHubOnCarrier(activeDomain, new Hub(registryHubTopStack.client, Scope.clone(registryHubTopStack.scope)));
+    if (!hasHubOnCarrier(activeDomain) || isOlderThan(getHubFromCarrier(activeDomain), API_VERSION)) {
+      const registryHubTopStack = getStackTop(getHubFromCarrier(registry));
```

Comment:
Basically, moved all the methods outside the class, and pass `hub` as the first parameter.